doc/developer-guides/asm_coding_guidelines.rst

```diff
@@ -0,0 +1,772 @@
+    .code32
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+          .extern      cpu_primary_save32
+          .extern   cpu_primary_save64
+       
+       .section     multiboot_header, "a"
+       .align  4
+       .long     0x0008
+       .long   0x0018
+       
+          .section   entry, "ax"
+          .align   8
+         .code32
+
+
+ASM-CS-03: Labels shall be aligned to the start of the line
+===========================================================
+
+Compliant example::
+
+    asm_showcase_1:
+            movl    $0x1, %eax
+    
+    asm_showcase_2:
+            movl    $0x2, %eax
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+         asm_showcase_1:
+            movl    $0x1, %eax
+       
+          asm_showcase_2:
+            movl    $0x2, %eax
+
+
+ASM-CS-04: Instruction statements shall be aligned
+==================================================
+
+An instruction statement is composed of instruction prefix, instruction
+mnemonic, and instruction operands. Instruction prefix and instruction operands
+are optional depending on the use case.
+Some detailed rules about the alignment are listed below:
+
+a) The start of instruction statements shall be aligned with one tab if the
+   instruction statement is in the code block under any label from functional
+   perspective. Otherwise, the start of instruction statements shall be aligned to
+   the start of the line. The start of the instruction could either be the
+   instruction mnemonic or the instruction prefix.
+b) One space shall be used to separate the instruction prefix and the
+   instruction mnemonic, where applicable.
+c) Tabs shall be used to separate the instruction mnemonic and the first
+   instruction operand, where applicable. The number of tabs could be decided by
+   the developers based on each case and it shall guarantee that the first
+   instruction operand in the code block under one label is aligned.
+d) A single space shall be used to separate multiple operands.
+
+Compliant example::
+
+    asm_showcase_1:
+            movl            $0x1, %eax
+            lock and        %rcx, (%rdx)
+    
+    asm_showcase_2:
+            movl            $0x3, %eax
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       asm_showcase_1:
+       movl   $0x1, %eax
+         lock    and        %rcx, (%rdx)
+       
+       asm_showcase_2:
+           movl     $0x2, %eax
+
+
+ASM-CS-05:  '//' shall not be used for comments
+===============================================
+
+'/* \*/' shall be used to replace '//' for comments.
+
+Compliant example::
+
+    /* This is a comment */
+    movl    $0x1, %eax
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       // This is a comment
+       movl    $0x1, %eax
+
+
+ASM-CS-06: Tabs shall be 8 characters wide
+==========================================
+
+A tab character shall be considered 8-character wide when limiting the line
```

Comment:
I don't see mention of a line length limit (80 or 120 perhaps)?

Reply:
> I don't see mention of a line length limit (80 or 120 perhaps)?

@dbkinder Thanks for pointing out!
Just added this rule in new version. 
Could you please help review it again?
Thanks!